R/join_glottolog_LanguageTable_ValueTable.R

```diff
@@ -0,0 +1,29 @@
+#' Joins LanguageTable and ValueTable
+#'
+#' @param LanguageTable data-frame of a CLDF LanguageTable
+#' @param ValueTable data-frame of a CLDF ValueTable
+#' @note The ValueTable is made wide using the columns "Language_ID", "Parameter_ID" and "Value". All other columns from the ValueTable are dropped.
+#' @return Data-frame of joined tables.
+#' @export
+
+join_LanguageTable_ValueTable <- function(LanguageTable = NULL,
+                                          ValueTable = NULL){
+
+    Table <- ValueTable %>%
+        reshape2::dcast(Language_ID ~ Parameter_ID, value.var = "Value") %>%
+        rename(ID = Language_ID) %>%
+        full_join(LanguageTable)
```

Comment:
I don't really see the utility of this function. Doing this join isn't particularly difficult or error prone, and whether the reshaping adds much value is presumably heavily dependent on the use case.

Reply:
Yes, good point. It was mainly because I figured what with all the references in the other functions to "you might wanna use glottolog-cldf" this might come in handy. Also, [Olena asked for help with something a bit related](https://github.com/grambank/rgrambank/issues/8) so I figured it might be useful. I find reshape::dcast/dplyr::spread()/dplyr::pivot_wider() okay to use but I know people struggle. 

It's a suggestion. I think there's some use for it, but not massively

Reply:
I think the `reshape` part is also somewhat problematic. It's another instance of the function doing too much. Why would a function named "join_*" also reshape the data? If the reshape part is left out, all that remains is a simple `full_join` - and I think if people struggle with that, they need to learn it rather than expect a package to implement such functionality for them.

Also note that it is not true in general that CLDF StructureDatasets have only one value per (Parameter, Language) (see for example APiCS).

Reply:
Okay, let's scrap this one. I know some wanted something like this, but there's too much and like you say it'd be rough with for example apics